rust/nasl-interpreter/src/built_in_functions/ssh.rs

```diff
@@ -0,0 +1,2107 @@
+///  
+/// nasl params
+///  
+/// - An SSH session id.
+///  
+/// nasl named params
+///  
+/// - login: A string with the login name.
+///  
+/// return A data block on success or NULL on error.
+fn nasl_ssh_login_interactive<K>(
+    register: &Register,
+    ctx: &Context<K>,
+) -> Result<NaslValue, FunctionErrorKind> {
+    let positional = register.positional();
+    if positional.is_empty() {
+        return Err(FunctionErrorKind::MissingPositionalArguments {
+            expected: 0,
+            got: 1,
+        });
+    }
+
+    let session_id = match &positional[0] {
+        NaslValue::Number(x) => *x as i32,
+        _ => {
+            return Err(FunctionErrorKind::WrongArgument(
+                ("Invalid session ID").to_string(),
+            ))
+        }
+    };
+
+    let login = match register.named("login") {
+        Some(ContextType::Value(NaslValue::String(x))) => Some(x.to_owned()),
+        _ => return Err(FunctionErrorKind::from("login")),
+    };
+
+    let mut sessions = lock_sessions(&ctx.sessions().ssh_sessions)?;
+    match sessions
+        .iter_mut()
+        .enumerate()
+        .find(|(_i, s)| s.session_id == session_id)
+    {
+        Some((_i, session)) => {
+            if !session.user_set {
+                set_opt_user(session, login, session_id)?;
+            }
+            let verbose = session.verbose > 0;
+
+            // Get the authentication methods only once per session.
+            let methods: AuthMethods = {
+                if !session.authmethods_valid {
+                    get_authmethods(session, session_id, ctx.logger())?
+                } else {
+                    session.authmethods
+                }
+            };
+            if verbose {
+                ctx.logger()
+                    .info(&format!("Available methods:\n{:?}", methods));
+            }
+            if methods.contains(AuthMethods::INTERACTIVE) {
+                let mut prompt = String::new();
+                loop {
+                    match session.session.userauth_keyboard_interactive(None, None) {
+                        Ok(AuthStatus::Info) => {
+                            let info = match session.session.userauth_keyboard_interactive_info() {
+                                Ok(i) => i,
+                                Err(_) => {
+                                    return Err(FunctionErrorKind::Diagnostic(
+                                        format!(
+                                            "Failed setting user authentication for SessionID {}",
+                                            session_id
+                                        ),
+                                        Some(NaslValue::Null),
+                                    ));
+                                }
+                            };
+                            if verbose {
+                                ctx.logger().info(&format!("SSH kbdint name={}", info.name));
```

Comment:
because this is use an ENV variable to switch it on, different from the general debug level. It will only be available if it is set via, which also turns the libssh debug log on     
```
let verbose = env::var("OPENVAS_LIBSSH_DEBUG")